### PR TITLE
README: improve docs for dependencies

### PR DESCRIPTION
**Pull request by arijit79**
Improved the dependencies section in the README. Also removed `gles3.2` dependency as it is usually provided by mesa

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -17,16 +17,24 @@ Hyprland's simple, yet multi-threaded and GPU-accelerated screen locking utility
 ## Building
 
 ### Deps
+Hyprlock requires `hyprlang` >= 0.4.0. See the 
+[README](https://github.com/hyprwm/hyprlang/blob/main/README.md#building-and-installation) for 
+build instructions.
```
Comment (vaxerski) [inline README.md:22]: why this part? Most package managers are now on >= 0.4.0

Comment (arijit79) [reply README.md:22]: Yeah I think the build from source part wasn't actually required as it is already shipped by default in most distros repos. Maybe we can rephrase it to say that if its not shipped on any distro's repos, people can still jump to its build instructions. I will put it in my next commit.

Comment (vaxerski) [reply README.md:22]: that's default for any dep, I think the comment is redundant.

Comment (arijit79) [reply README.md]: Ok. I changed it in my latest commit.

File: README.md [hunk 2]
```diff
@@ -17,20 +17,17 @@ Hyprland's simple, yet multi-threaded and GPU-accelerated screen locking utility
...
- And the development libraries for the following:-
- - cairo
- - libdrm
- - pango
- - xkbcommon
- - pam
+- wayland-client
+- wayland-protocols
+- mesa
+And the development libraries for the following:-
```
Comment (vaxerski) [inline README.md:24]: why is there a - at the end here xd

**Review by vaxerski — COMMENTED** [review]
rest lgtm

File: README.md (hunk 2 above)
Comment (arijit79) [reply README.md:24]: I just like to add a - in these types of lines, just have a habit of it. I will remove that if you don't like that.

Comment (vaxerski) [reply README.md:24]: ye please dont

**Review by vaxerski — APPROVED** [review]
